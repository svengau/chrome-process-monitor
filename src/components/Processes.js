import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import bytes from 'bytes';
import styled from 'styled-components';

// import Logging from './Logging';

const Table = styled.table`
  display:table;
  width:100%;
  border-collapse:collapse;
  font-size: 12px;
  font-family: Lato,'Helvetica Neue',Arial,Helvetica,sans-serif;
`;

const Task = styled.td`
  display: table-cell;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 150px;
  width:100%;
`;

const Number = styled.td`
  text-align: right;
  padding: 0 10px;
`;

const Process = styled.td`
  text-align: left;
  padding: 0 10px;
`;

const Th = styled.th`
  font-weight: bold;
  padding: 0 10px;
`;

const Loading = styled.div`
  width: 100%;
  text-align: center;
  vertical-align: middle;
  line-height: 360px;
  height: 360px;
`;

const GithubLink = styled.div`
  text-align: right;
  width: 100%;
  vertical-align: middle;
  text-decoration: none;
  color: black;
  line-height: 18px;
  height: 18px;
  font-family: Lato,'Helvetica Neue',Arial,Helvetica,sans-serif;
  cursor: pointer;
`;

const GithubImg = styled.img`
  vertical-align: middle;
  height: 18px;
  width: 18px;
  margin-bottom: 3px;
  margin-right: 10px;
  margin-left: 5px;
`;

/**
 * Doc: https://developer.chrome.com/extensions/processes
 */
class Processes extends React.PureComponent {
  goToGithub = () => {
    if (chrome && chrome.tabs) {
      chrome.tabs.create({ url: 'https://github.com/svengau/chrome-process-monitor' });
    }
  };

  render() {
    const { currentProcessId, processes } = this.props;

    // Logging.info('Processes::render', { currentProcessId, processes });

    if (!processes || processes.length === 0) {
      return (
        <Loading>
          Loading...
        </Loading>);
    }

    return (
      <Fragment>
        <Table>
          <thead>
            <tr>
              <Th>Process ID</Th>
              <Th>Tab</Th>
              <Th>CPU</Th>
              <Th>Private memory</Th>
              <Th>Javascript memory allocated</Th>
              <Th>Javascript memory used</Th>
            </tr>
          </thead>
          <tbody>
            {processes.map(process => (
              <tr key={process.id} style={{ backgroundColor: (currentProcessId === process.id) ? 'orange' : '' }}>
                <Process>{process.osProcessId}</Process>
                <Task>{process.tasks && process.tasks[0] && process.tasks[0].title}</Task>
                <Number>{process.cpu ? Math.round(process.cpu) : '-'}</Number>
                <Number>{process.privateMemory ? bytes(process.privateMemory) : '-'}</Number>
                <Number>{process.jsMemoryAllocated ? bytes(process.jsMemoryAllocated) : '-'}</Number>
                <Number>{process.jsMemoryUsed ? bytes(process.jsMemoryUsed) : '-'}</Number>
              </tr>
          ))}
          </tbody>
        </Table>
        <GithubLink onClick={this.goToGithub}>
            Hosted on Github <GithubImg alt="github" src="assets/github-32.png" />
        </GithubLink>
      </Fragment>
    );
  }
}

Processes.propTypes = {
  currentProcessId: PropTypes.number,
  processes: PropTypes.array,
};

Processes.defaultProps = {
  currentProcessId: null,
  processes: [],
};

export default Processes;
