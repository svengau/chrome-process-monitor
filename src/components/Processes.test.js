/* global it, expect */
import React from 'react';
import renderer from 'react-test-renderer';
import Processes from './Processes';
import processes from './processes.fixtures';

it('renders empty processes correctly', () => {
  const tree = renderer
    .create(<Processes />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a list of processes correctly', () => {
  const tree = renderer
    .create(<Processes processes={processes} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
