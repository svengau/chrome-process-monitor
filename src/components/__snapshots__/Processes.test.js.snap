// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of processes correctly 1`] = `
Array [
  <table
    className="sc-bdVaJa kxmuZk"
  >
    <thead>
      <tr>
        <th
          className="sc-ifAKCX jbVedC"
        >
          Process ID
        </th>
        <th
          className="sc-ifAKCX jbVedC"
        >
          Tab
        </th>
        <th
          className="sc-ifAKCX jbVedC"
        >
          CPU
        </th>
        <th
          className="sc-ifAKCX jbVedC"
        >
          Private memory
        </th>
        <th
          className="sc-ifAKCX jbVedC"
        >
          Javascript memory allocated
        </th>
        <th
          className="sc-ifAKCX jbVedC"
        >
          Javascript memory used
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86348
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Browser
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          64
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          177.36MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          -
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          -
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86352
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          GPU Process
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          1
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          40.79MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          -
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          -
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86367
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Extension: Web Developer
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          0
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          9.46MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          9.85MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          6.56MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86377
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Extension: WhatRuns
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          0
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          54.76MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          15.37MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          10.76MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86383
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Google Maps
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          0
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          85.86MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          89.25MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          71.41MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86384
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Subframe: AdBlock
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          1
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          190.6MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          152.69MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          139.93MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86402
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Tab: Inbox â€“ xxxxxx@gmail.com
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          1
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          295.59MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          213.33MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          193.38MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86419
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Tab: * Feature Requests | Trello
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          0
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          112.46MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          77.26MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          66.27MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86442
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Tab: * Bugs | Trello
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          0
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          87.45MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          63.26MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          54.04MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86448
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Tab: TODO - Wunderlist
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          1
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          49.99MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          53.21MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          42.26MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86456
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Tab: Microsoft
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          0
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          26.05MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          16.86MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          10.14MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86406
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Tab: chrome.processes - Google Chrome
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          1
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          60.08MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          29.21MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          22.3MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86371
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Tab: Extensions
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          1
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          185.69MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          21.85MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          13.26MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          86427
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Subframe: https://accounts.google.com/
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          1
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          65.84MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          36.86MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          22.21MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          97713
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Subframe: React Developer Tools
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          1
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          30.64MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          13.85MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          8.93MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          813
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Tab: My tracks - Artistco
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          0
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          136.42MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          64.21MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          54.62MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          907
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Tab: Developer Tools - https://www.artistco.com/me/tracks
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          1
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          104.51MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          77.34MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          65.54MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          97647
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Tab: Developer Tools - chrome://extensions/
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          1
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          84.21MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          39.7MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          34.78MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          19900
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Tab: Developer Tools - chrome-extension
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          3
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          72.32MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          23.41MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          17MB
        </td>
      </tr>
      <tr
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        <td
          className="sc-bxivhb bgPqam"
        >
          21892
        </td>
        <td
          className="sc-bwzfXH ftKEZi"
        >
          Tab: React.Component - React
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          0
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          83.98MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          27.36MB
        </td>
        <td
          className="sc-htpNat kSWrKi"
        >
          20.62MB
        </td>
      </tr>
    </tbody>
  </table>,
  <div
    className="sc-bZQynM hLMFvz"
    onClick={[Function]}
  >
    Hosted on Github 
    <img
      alt="github"
      className="sc-gzVnrw bcVbOd"
      src="assets/github-32.png"
    />
  </div>,
]
`;

exports[`renders empty processes correctly 1`] = `
<div
  className="sc-EHOje hJHkPF"
>
  Loading...
</div>
`;
